src/common/mfu_flist_walk.c

```diff
@@ -506,10 +524,12 @@ static void walk_stat_process(CIRCLE_handle* handle)
     /* get path from queue */
     char path[CIRCLE_MAX_STRING_LEN];
     handle->dequeue(path);
+    mfu_file_t* mfu_file = *pfile;
 
     /* stat item */
     struct stat st;
-    int status = mfu_lstat(path, &st);
+    int status;
+    status = mfu_file_lstat(path, &st, mfu_file);
```

Comment:
More thinking out loud.  Do we want to declare the mfu_file type here or deduce it from the path on the fly?